src/share/classes/com/sun/javatest/regtest/tool/ExcludeFileVerifier.java

```diff
@@ -0,0 +1,150 @@
+/*
+ * Copyright (c) 2024, Oracle and/or its affiliates. All rights reserved.
+ * DO NOT ALTER OR REMOVE COPYRIGHT NOTICES OR THIS FILE HEADER.
+ *
+ * This code is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU General Public License version 2 only, as
+ * published by the Free Software Foundation.  Oracle designates this
+ * particular file as subject to the "Classpath" exception as provided
+ * by Oracle in the LICENSE file that accompanied this code.
+ *
+ * This code is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+ * FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License
+ * version 2 for more details (a copy is included in the LICENSE file that
+ * accompanied this code).
+ *
+ * You should have received a copy of the GNU General Public License version
+ * 2 along with this work; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301 USA.
+ *
+ * Please contact Oracle, 500 Oracle Parkway, Redwood Shores, CA 94065 USA
+ * or visit www.oracle.com if you need additional information or have any
+ * questions.
+ */
+
+package com.sun.javatest.regtest.tool;
+
+import java.io.BufferedReader;
+import java.io.FileReader;
+import java.io.File;
+import java.io.FileNotFoundException;
+import java.io.IOException;
+import java.io.PrintWriter;
+import java.nio.file.Path;
+import java.util.ArrayList;
+import java.util.function.Predicate;
+import java.util.List;
+import java.util.regex.Pattern;
+
+public class ExcludeFileVerifier {
+    private PrintWriter out;
+
+    public ExcludeFileVerifier(PrintWriter out) {
+        this.out = out;
+    }
+
+    private boolean hadErrors = false;
+    public boolean getHadErrors() {
+        return hadErrors;
+    }
+    public boolean verify(File file, List<String> validTestNames) {
+        ArrayList<String> usedTestNames = new ArrayList<String>();
+        ArrayList<Check> checks = new ArrayList<Check>();
```

Comment:
If old school, it would be more appropriate to use just `List` on the LHS, if that is sufficient.

If new school, use `var`.

Either way, you don't need duplicate type parameters. Either use diamond syntax or `ver`.